packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -115,198 +118,316 @@ describeNoCompat("GC DataStore Tombstoned When It Is Sweep Ready", (getTestObjec
+            },
+            `Should not be able to retrieve a tombstoned datastore.`,
+        );
+    });
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sweepTimeoutMs` above
+    itExpects("Requesting tombstoned datastores fails in summarizing container loaded before sweep timeout",
+    [
+        {
+            error: "TombstonedDataStoreRequested",
+            eventName: "fluid:telemetry:ContainerRuntime:TombstonedDataStoreRequested",
+            viaHandle: false,
+        },
+    ],
+    async () => {
+        const {
+            unreferencedId,
+            summarizingContainer,
+            summarizer,
+        } = await summarizationWithUnreferencedDataStoreAfterTime(sweepTimeoutMs - waitLessThanSweepTimeoutMs);
+        // Wait enough time so that the datastore is sweep ready
+        await delay(waitLessThanSweepTimeoutMs);
+
+        await sendOpToUpdateSummaryTimestampToNow(summarizingContainer);
+
+        // The datastore should be tombstoned now
+        await summarize(summarizer);
+
+        // Requesting a tombstoned datastore should fail!
+        await assert.rejects(async () => requestFluidObject<ITestDataObject>(summarizingContainer, unreferencedId),
+            (error) => {
+                const correctErrorType = error.code === 404;
+                const correctErrorMessage = error.message.startsWith(`Datastore removed by gc`) === true;
+                return correctErrorType && correctErrorMessage;
+            },
+            `Should not be able to retrieve a tombstoned datastore.`,
+        );
+    });
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sweepTimeoutMs` above
+    itExpects("Handle request for tombstoned datastores fails in summarizing container loaded after sweep timeout",
+    [
+        { eventName: "fluid:telemetry:Summarizer:Running:SweepReadyObject_Loaded" },
+        {
+            error: "TombstonedDataStoreRequested",
+            eventName: "fluid:telemetry:ContainerRuntime:TombstonedDataStoreRequested",
+            viaHandle: true,
+        },
+    ],
+    async () => {
+        const {
+            unreferencedId,
+            summarizingContainer,
+            summarizer,
+        } = await summarizationWithUnreferencedDataStoreAfterTime(sweepTimeoutMs);
+
+        const dataObject = await requestFluidObject<ITestDataObject>(summarizingContainer, unreferencedId);
+        await sendOpToUpdateSummaryTimestampToNow(summarizingContainer);
+
+        // The datastore should be tombstoned now
+        await summarize(summarizer);
+
+        // Note: if a user makes a request that looks like this, we will also think that the request is via handle
+        const request: IRequest = { url: unreferencedId, headers: { [RuntimeHeaders.viaHandle]: true } };
+        const response = await dataObject._context.IFluidHandleContext.resolveHandle(request);
```

Comment:
I was just in a meeting all about the difference between `resolveHandle` and `handle.get`.  I am honestly confused.  Is there a way to create a handle and call get on it?  Is this what RemoteObjectFluidHandle is or whatever? Let's discuss elsewhere.